[TPS - 2404] Update Partner API documentation (#452)
zapier/visual-builder: docs/_embed/partner_api.md

Available parameters to the Apps resource:

| parameter                   | requirement | notes                                                                                         |
| --------------------------- | ----------- | --------------------------------------------------------------------------------------------- |
| **client_id**               | Required    | Your application client ID.                                                                   |

> | **category**                | Optional    | Filter the results by app category.                                                           |

Reviewer: Should we provide guidance on what valid options are available for this category param? As a Partner, I don't think I'd be aware of all the options.

> | **category**                | Optional    | Filter the results by app category.                                                           |

Reviewer: Can we specify if this param is expecting the category title or slug? It's not super obvious to me with the google example.

Reviewer: yep! i think mentioning slug will make it clear. TY~

> | **is_in_zap_template_with** | Optional    | Fetch apps that appear in a Zap Template with your app. |

Reviewer: It might make it more clear there's no value expected for this parameter like all the others by denoting in the notes "No value required for this param." or something like that.

Author: What about `Fetch apps that appear in a Zap Template with your app (No value required).`?

Reviewer: Fetch apps that appear in a Zap Template with your app. No value required.
or
Fetch apps that appear in a Zap Template with your app (no value required).

Minor formatting suggestions^

> | **is_in_zap_template_with** | Optional    | Service slug that limits the results to apps that are included within the same Zap Templates. |

Reviewer 2: The partner does not need to specify a service slug - it doesn't matter what they specify, so long as `is_in_zap_template_with` is in the query params. I would make the notes something like

```suggestion
| **is_in_zap_template_with** | Optional    | Fetch apps that appear in a Zap Template with your app. |
```

Reviewer 2: Looks like this got marked as resolved without changing the notes.

Author: So what happened was I used git's Commit Suggestions and then later accidentally forced pushed over those changes. Should be good now.

> | **is_in_zap_template_with** | Optional    | Service slug that limits the results to apps that are included within the same Zap Templates. |

Author: @cyberwitch Did we change the name of this to `shares_zap_templates` recently?

Reviewer 2: Oops, that's an old name. Will update it - thanks for spotting the discrepancy!

> | **title_search**            | Optional    | Filter the results by matching title (case-insensitive).                                      |

Author: Users can only provide one title there a curl example below but maybe it's not clear? Couple options would be to. Mention it in parens like we do for other parameters like `Filter the results by matching title (case-insensitive single title)`. Alternatively add an extra column called "type" would provide better specificity.
So for the `title_search` case the type would be `string`.

> | **title_search**            | Optional    | Filter the results by matching title (case-insensitive).                                      |

Reviewer 2: To be clear, `title_search` will search for the string anywhere in the title, not just exact matches. So it can return multiple results - you can search for the letter "s" if you like!

But you have to do multiple searches to search for multiple different strings.

> | **title_search**            | Optional    | Filter the results by matching title (case-insensitive).                                      |

Author: Ok this sounds like it would be more accurate:
`Filter the results by performing a case-insensitive search for a substring within the app title.`

> | **title_search**            | Optional    | Filter the results by matching title (case-insensitive).                                      |

Reviewer: Thanks for the clarification! Could use following for a shorter clause: Filter the results with a case-insensitive search for a substring within the app title.

curl -L "https://api.zapier.com/v1/apps?client_id=${client_id}"
```

Get a list of apps related to google

```bash
curl -L "https://api.zapier.com/v1/apps?client_id=${client_id}&category=google"
```

> Get a list of apps that are included in the same Zap Template as the Google Calendar app
> 
> ```bash
> curl -L "https://api.zapier.com/v1/apps?client_id=${client_id}&is_in_zap_template_with=google-calendar"

Reviewer 2: ```suggestion
Get a list of apps that are included in the same Zap Template as your app

```bash
curl -L "https://api.zapier.com/v1/apps?client_id=${client_id}&is_in_zap_template_with"
```

```

Get the Google Calendar app

```bash

> curl -L "https://api.zapier.com/v1/apps?client_id=${client_id}&title_search=google%calendar"

Reviewer 2: I think we want either

```suggestion
curl -L "https://api.zapier.com/v1/apps?client_id=${client_id}&title_search=google+calendar"
```

or `title_search=google%20calendar` or `title_search=google calendar`.

Reviewer 2: It doesn't look like this was updated either

| parameter                   | requirement | notes                                                                                                                                      |
| --------------------------- | ----------- | ------------------------------------------------------------------------------------------------------------------------------------------ |
| **params\_\_{KEY}={VALUE}** | Optional    | Return Zaps that have a specific key/value set in the params (settings) of the Zap. |

> | **get_params** | Optional    | Return Zaps along with their associated key/value param settings. |

Reviewer 2: I noticed this wasn't added to the changelog. Feel free to remove this for now, or make a note that this MR should not be merged as is until https://zapierorg.atlassian.net/browse/TPS-2398 is Done.

Reviewer 2: The feature is merged, so all good here!

I would add an example of using it though - `/v1/zaps?get_params` will return a response where steps that belong to the partner's app have `"params": {"some_param": "some_value"}` while all other steps will have `"params": null"`.

Author: Updated the existing example to include the params field.

curl -H "Authorization: Bearer {token}" -L "https://api.zapier.com/v1/zaps?&params__board=BOARD_ID"
```

> Get all Zaps in the user's account and include their associated param key/value pairs (assuming the OAuth app is Trello).

Reviewer 2: ```suggestion
Get all Zaps in the user's account and include Trello steps' `params` key/value pairs (assuming the OAuth app is Trello).
```

Small suggestion for clarity. Otherwise, I think this is good to go (whenever @rjpowell says to release)!

```bash
curl -H "Authorization: Bearer {token}" -L "https://api.zapier.com/v1/zaps?&get_params"
```

**Example Response**

> Refer to the [Zap](#zap) reference for all the available fields.

```json
{
    "next": "https://api.zapier.com/v1/zaps?limit=2&offset=4",
    "previous": "https://api.zapier.com/v1/zaps?limit=2",
    "count": 20,
    "objects": [
        {
            "id": 159061877,

>             "title": "",

Reviewer 3: Is a blank `title` expected?

Author: @gregilo `/v1/zaps` is calls the `/v4/zaps/` endpoint. And AFAICT `/v4/zaps/` allowed blanks strings so it is to be expected.

Reviewer 3: @MagWeaver gotcha - thanks for looking into this further!

```

## Changelog
- 2023-05-25

  - Added query parameters to `v1/apps`

    - The endpoint `/v1/apps` supports query parameters `title_search`, `title_starts_with`, and `category` for more refined app search results

  - Added query parameter to `v1/zaps`

>     - The endpoint `/v1/zaps` supports the query parameters `is_in_zap_template_with`.

Reviewer 2: This will be under `/v1/apps`, not `/v1/zaps`. It's not yet finished either, so feel free to either leave it out of the MR for now, or make a note that this MR cannot be merged until https://zapierorg.atlassian.net/browse/TPS-2402 and https://zapierorg.atlassian.net/browse/TPS-2403 are Done.

Reviewer 2: Finished! Still should be under `/v1/apps` with the other new query params.

You can add `get_params` to `/v1/zaps` though!
